### PR TITLE
Read dataDirectory from system/env variables

### PR DESCRIPTION
**Pull request by pratyakshsharma**
Refactor IcebergQueryRunner and HiveQueryRunner to read the data directory from system variables rather than program arguments. This is done to ensure these runners are in sync with HiveExternalWorkerQueryRunner.

## Impact
No impact

## Test Plan
NA

## Contributor checklist

- [ ] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [ ] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [ ] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [ ] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [ ] Adequate tests were added if applicable.
- [ ] CI passed.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-iceberg/src/test/java/com/facebook/presto/iceberg/IcebergQueryRunner.java [hunk 1]
```diff
@@ -176,14 +177,9 @@ public static void main(String[] args)
...
         Optional<Path> dataDirectory = Optional.empty();
-        if (args.length > 0) {
-            if (args.length != 1) {
-                log.error("usage: IcebergQueryRunner [dataDirectory]\n");
-                log.error("       [dataDirectory] is a local directory under which you want the iceberg_data directory to be created.]\n");
-                System.exit(1);
-            }
-
-            File dataDirectoryFile = new File(args[0]);
+        Optional<String> suppliedDataDirectoryPath = getProperty("DATA_DIR");
```
Comment (yingsu00) [inline presto-iceberg/src/test/java/com/facebook/presto/iceberg/IcebergQueryRunner.java:180]: getProperty("DATA_DIR") would return Optional.empty() if the property is not set, so you don't need to use another variable suppliedDataDirectoryPath

File: presto-hive/src/test/java/com/facebook/presto/hive/HiveQueryRunner.java [hunk 2]
```diff
@@ -441,14 +442,9 @@ public static void main(String[] args)
...
         Optional<Path> dataDirectory = Optional.empty();
-        if (args.length > 0) {
-            if (args.length != 1) {
-                log.error("usage: HiveQueryRunner [dataDirectory]\n");
-                log.error("       [dataDirectory] is a local directory under which you want the hive_data directory to be created.]\n");
-                System.exit(1);
-            }
-
-            File dataDirectoryFile = new File(args[0]);
+        Optional<String> suppliedDataDirectoryPath = getProperty("DATA_DIR");
```
Comment (yingsu00) [inline presto-hive/src/test/java/com/facebook/presto/hive/HiveQueryRunner.java:445]: remove the variable suppliedDataDirectoryPath

File: presto-hive/src/test/java/com/facebook/presto/hive/HiveQueryRunner.java [hunk 3]
```diff
@@ -441,14 +442,9 @@ public static void main(String[] args)
...
+        if (suppliedDataDirectoryPath.isPresent()) {
+            File dataDirectoryFile = new File(suppliedDataDirectoryPath.get());
             if (dataDirectoryFile.exists()) {
```
Comment (yingsu00) [inline presto-hive/src/test/java/com/facebook/presto/hive/HiveQueryRunner.java:448]: Extract a common method to validate the data directory.

File: presto-iceberg/pom.xml [hunk 4]
```diff
@@ -505,6 +504,13 @@
             <scope>test</scope>
         </dependency>
 
+        <dependency>
```
Comment (yingsu00) [inline presto-iceberg/pom.xml:507]: Why is this change needed?

Comment (pratyakshsharma) [reply presto-iceberg/pom.xml:507]: This is needed to make use of HiveTestUtils.getProperty() method in IcebergQueryRunner.

File: presto-iceberg/src/test/java/com/facebook/presto/iceberg/IcebergQueryRunner.java (hunk 1 above)
Comment (pratyakshsharma) [reply presto-iceberg/src/test/java/com/facebook/presto/iceberg/IcebergQueryRunner.java:180]: I had introduced the variable so that we do not need to call getProperty twice, first on L181 and then on L182.

File: presto-hive/src/test/java/com/facebook/presto/hive/HiveQueryRunner.java (hunk 3 above)
Comment (pratyakshsharma) [reply presto-hive/src/test/java/com/facebook/presto/hive/HiveQueryRunner.java]: Done.

File: presto-iceberg/src/test/java/com/facebook/presto/iceberg/IcebergQueryRunner.java [hunk 5]
```diff
@@ -282,39 +282,7 @@ public static void main(String[] args)
...
-                    dataDirectoryFile = dataDirectoryFile.getParentFile();
-                }
-                if (!dataDirectoryFile.canRead() || !dataDirectoryFile.canWrite()) {
-                    log.error("Error: The ancestor directory " + dataDirectoryFile.getAbsolutePath() + " is not readable/writable.");
-                    System.exit(1);
-                }
-            }
-
-            dataDirectory = Optional.of(dataDirectoryFile.toPath());
-        }
```
Comment (ZacBlanco) [inline presto-iceberg/src/test/java/com/facebook/presto/iceberg/IcebergQueryRunner.java:317]: I don't really see why we need to remove the ability to specify the data directory through program arguments. I have good number of IDE run configurations which utilize the program arguments already. Can't we just add the ability to read from the env variables if program arguments aren't specified?

In my experience designing an application that takes some number of configuration values, they should be resolved in the following priority (higher options take precedence over lower ones)

1. CLI arguments
2. environment variables
3. user config: e.g. read from `$HOME/.config` or similar
4. system config e.g. `/etc/....`

Comment (pratyakshsharma) [reply presto-iceberg/src/test/java/com/facebook/presto/iceberg/IcebergQueryRunner.java:317]: Makes sense, will do the changes.

**Review by ZacBlanco — APPROVED** [review]

**Review by yingsu00 — APPROVED** [review]
